models/skilledImpactRequestsModel.js

```diff
@@ -0,0 +1,73 @@
+const db = require('../database/dbConfig');
+
+const ProjectGoal = require('./projectGoalsModel.js');
+
+
+async function findSkilledImpactRequests(campaign_id) {
+    return db('project_goals')
+        .where({campaign_id})
+        .fullOuterJoin('skilled_impact_requests', 'skilled_impact_requests.id','project_goals.skilled_impact_request_id')
+        .select(
+           '*'
+        )
+        .then((returnedQuery)=>{
+            const skilledRequests = new Map();
+            for(let i = 0; i<returnedQuery.length; i++){
+                if(!skilledRequests.has(returnedQuery[i].id)){
+                    let skillAndProject = {
+                        skill: returnedQuery[i].skill,
+                        point_of_contact: returnedQuery[i].point_of_contact,
+                        welcome_message: returnedQuery[i].welcome_message,
+                        our_contribution: returnedQuery[i].our_contribution,
+                        project_goals:[]
+                    };
+                    skilledRequests.set(returnedQuery[i].id, skillAndProject);
+                }
+                let projectGoal = {
+                    goal_title: returnedQuery[i].goal_title,
+                    description: returnedQuery[i].description,
+                };
+                skilledRequests.get(returnedQuery[i].id).project_goals.push(projectGoal);
+            }
+            return Array.from(skilledRequests.values());
+        });
+}
+
+
+async function insertSkilledRequestsAndProjectGoals(skilledRequests, campaign_id){
+    let skillImpactRequests={};
+
+    for (let skilledRequest of skilledRequests) {
```

Comment:
Yes but in this case though we are adding the campaign_id props into the JSON, if it is constant we cannot add it. The reason we need to add it here is because campaign gets inserted before this and then the id for that campaign gets returned which is passed through the parameter. I assume the campaign_id prop will not exist in the req.body JSON for each skilledImpactRequests because we do not know the ID before inserting them. However, if there is an easier way than this, I am completely okay with changing it.